### DISCUSSION TITLE
insta Serialize Box

### DISCUSSION THREAD

**Discussion by DasLixou** [mitsuhiko/insta#395] [Q&A]
So I have a trait and have a `Box<dyn MyTrait>` of that trait. now i want to snapshot test this box with `assert_ron_snapshot!`. But that doesnt work because it doesnt know that the thing behind MyTrait is Serializable. I also tried to make `trait MyTrait: Serializable` but then it can't be made into an object. Any suggestions?

Also: Is there a way to make the tests back into a mod tests { } ? currently i have to make them out of that #[cfg(debug)] flagged module for insta to see them. Can i also change the snapshot path to not be inside the src/ folder?

**Comment by max-sixty** [comment]
Late reply but — insta tries to serialize the object, but otherwise the object is opaque to insta. The standard approach would be to unbox it?
